workflows/flows/copy_amplicon_pipeline_results.py

```diff
@@ -0,0 +1,121 @@
+from prefect import flow, get_run_logger
+import django
+from pathlib import Path
+django.setup()
+
+from analyses.models import Analysis
+from workflows.data_io_utils.mgnify_v6_utils.amplicon import EMG_CONFIG
+from workflows.prefect_utils.slurm_flow import move_data
+
+
+@flow(
+    name="Copy Amplicon Pipeline Results",
+    flow_run_name="Copy Amplicon Pipeline Results",
+    log_prints=True,
+)
+# async def copy_amplicon_pipeline_results(analysis: Analysis):
+def copy_amplicon_pipeline_results(analysis_accession: str):
+    analysis = Analysis.objects.get(accession=analysis_accession)
+    # logger = get_run_logger()
+    print(f"Copying Amplicon Pipeline Results for {analysis.accession}")
+    print(f"Results dir: {analysis.results_dir}")
+
+    print(f"FTP results dir: {EMG_CONFIG.slurm.ftp_results_dir}")
+
+    print(f"Study first accession: {analysis.study.accession}")
+    # move_data('/nfs/production/rdf/metagenomics/results', '/nfs/ftp/public/databases/metagenomics/mgnify_results',
+    #           'cp -r')
+    # move_data(f"/nfs/production/rdf/metagenomics/results/{analysis.results_dir}",
+    #     f"{EMG_CONFIG.slurm.ftp_results_dir}/{analysis.study.accession[:4]}/{analysis.study.accession}",
+    #     'cp -r'
+    # )
+
+    source = f"/nfs/production/rdf/metagenomics/results/{analysis.results_dir}"
+    target = f"{EMG_CONFIG.slurm.ftp_results_dir}/{analysis.study.accession[:4]}/{analysis.study.accession}"
+
+    # ALLOWED_EXTENSIONS = {
+    #     '.yml',  # software_versions.yml
+    #     '.txt', '.tsv',  # taxonomy and ASV files
+    #     '.mseq',  # taxonomy files
+    #     '.html',  # reports
+    #     '.fa', '.fasta',  # sequence files
+    #     '.gz',  # compressed fastq
+    #     '.json',  # metadata and stats
+    #     '.csv',  # QC files
+    # }
+
+    # ALLOWED_PATTERNS = {
+    #     'pipeline_info': ['software_versions.yml'],
+    #     'taxonomy-summary': ['*.txt', '*.tsv', '*.mseq', '*.html'],
+    #     'sequence-categorisation': ['*.fa', '*.fasta', '*.deoverlapped'],
+    #     'qc': ['*.html', '*.json', '*.tsv', '*.gz'],
+    #     'primer-identification': ['*.tsv', '*.fasta', '*.json'],
+    #     'asv': ['*.tsv', '*.fasta'],
+    #     'amplified-region-inference': ['*.tsv', '*.txt'],
+    #     '.': ['sample.txt','study_multiqc_report.html', 'qc_*.csv', 'primer_*.json', 'manifest.json']
+    # }
+
+    ALLOWED_PATTERNS = {
```

Comment:
Why can't we do all folder and file creation with python?

Reply:
@mberacochea I've changed how this works please have a look.
this part of the code does not actually move the files. 
It just checks for invalid files and then calls the move_data utility. 
which in turn triggers a slurm job.